src/js/components/Carousel/Carousel.js

```diff
@@ -136,13 +137,33 @@ const Carousel = ({
       animation = { type: 'fadeOut', duration: 0 };
     }
 
-    return (
-      <Box fill={fill} overflow="hidden">
-        <Box fill={fill} animation={animation}>
-          {child}
-        </Box>
-      </Box>
-    );
+    if (index === activeIndex) {
+      wrappedChildren.push(
+        // eslint-disable-next-line react/no-array-index-key
+        <Box key={index} fill={fill} overflow="hidden" style={{ zIndex: 2 }}>
+          <Box fill={fill} animation={animation}>
+            {child}
+          </Box>
+        </Box>,
+      );
+    } else {
+      wrappedChildren.unshift(
+        <Box
+          // eslint-disable-next-line react/no-array-index-key
+          key={index}
+          fill={fill}
+          overflow="hidden"
+          style={{
+            zIndex: index === priorActiveIndex ? 1 : 0,
```

Comment:
Same as above, do we need `zIndex`/`opacity`? Is there any way to reorder the children like you're doing and leverage the existing way that the other children's visibility was controlled?